operator/src/lib.rs

```diff
@@ -48,15 +52,24 @@ use strum::IntoEnumIterator;
 use tracing::{debug, info, trace, warn};
 
 const FINALIZER_NAME: &str = "nifi.stackable.tech/cleanup";
+const SHOULD_BE_SCRAPED: &str = "monitoring.stackable.tech/should_be_scraped";
+
+const HTTP_PORT_NAME: &str = "http";
+const PROTOCOL_PORT_NAME: &str = "protocol";
+const LOAD_BALANCE_PORT_NAME: &str = "loadbalance";
+const METRICS_PORT_NAME: &str = "metrics";
 
-type NifiReconcileResult = ReconcileResult<error::Error>;
+const CONTAINER_NAME: &str = "nifi";
+
+type NifiReconcileResult = ReconcileResult<error::NifiError>;
 
 struct NifiState {
     context: ReconciliationContext<NifiCluster>,
+    eligible_nodes: EligibleNodesForRoleAndGroup,
     existing_pods: Vec<Pod>,
-    eligible_nodes: HashMap<String, HashMap<String, Vec<Node>>>,
-    zookeeper_info: Option<ZookeeperConnectionInformation>,
+    monitoring: Arc<MonitoringStatus>,
```

Comment:
`MonitoringStatus` is completely stateless http client and some headers and it is only used in the function `enable_monitoring()`. It doesn't need to be part of the NifiStatus at all. I think it can be a local variable inside the `enable_monitoring()` closure.

Reply:
As discussed this is a question between creating the client over and over or doing it once. Right now it is once, but i think to simplify, created it with every reconcile is ok too.